src/pages/book-detail/BookDetailPage.tsx

```diff
@@ -0,0 +1,16 @@
+import { useParams } from 'react-router-dom'
+import { useBookInfoQuery, type BookInfo } from '@/hooks/queries/useBookInfo'
+import BookDetail from '@/components/BookDetail/BookDetail'
+import SidebarRecommendations from '@/components/BookDetail/SidebarRecommendations'
+
+export default function BookDetailPage() {
+  const { isbn } = useParams()
```

Comment:
[REQ] We should handle this cases too with something like this
```suggestion
  const { isbn } = useParams()
  if (!isbn) return <div> ISBN is not valid </div>
  const bookInfoQuery = useBookInfoQuery(isbn)
  if (bookInfoQuery.isLoading) return <div> Loading... </div>
  const bookInfo = bookInfoQuery.data || ({} as BookInfo)
  return (
    <div className="flex flex-row relative">
      <BookDetail isbn={isbn} bookInfo={bookInfo} />
      <SidebarRecommendations genre={''} />
    </div>
  )
```

Reply:
That modification throws this error in line 9 `React Hook "useBookInfoQuery" is called conditionally. React Hooks must be called in the exact same order in every component render.eslint[react-hooks/rules-of-hooks](https://reactjs.org/docs/hooks-rules.html)`, apparently it's not possible to put conditionals before hooks 🤔

Reply:
Maybe we could do this? Do you know a better way to solve this?
```suggestion
  const { isbn } = useParams()
  const bookInfoQuery = useBookInfoQuery(isbn || '')
  if (!isbn) return <div> ISBN is not valid </div>
  const bookInfo = bookInfoQuery?.data || ({} as BookInfo)
  if (bookInfoQuery.isLoading) return <div> Loading... </div>
  return (
    <div className="flex flex-row relative">
      <BookDetail isbn={isbn || ''} bookInfo={bookInfo} />
      <SidebarRecommendations genre={''} />
    </div>
  )
}
```

Reply:
Good point, I didn't notice that error. And what do you think about doing it like this? This way we encapsulate more the logic of BookDetail and we only trigger the query of BookInfo if the isbn is a valid string.
```suggestion
  const { isbn } = useParams()
  if (!isbn) return <div> ISBN is not valid </div>
  return <BookDetailSection isbn={isbn} />
}

const BookDetailSection = ({ isbn }: { isbn: string }) => {
  const bookInfoQuery = useBookInfoQuery(isbn)
  if (bookInfoQuery.isLoading) return <div> Loading... </div>
  const bookInfo = bookInfoQuery.data || ({} as BookInfo)
  return (
    <div className="flex flex-row relative">
      <BookDetail isbn={isbn} bookInfo={bookInfo} />
      <SidebarRecommendations genre={''} />
    </div>
  )
}

```
Please, let me know your thoughts!

Reply:
We could also check if the isbn is a valid string too, please [take a look at this](https://www.geeksforgeeks.org/program-check-isbn/). Then we can have something like:

```
export default function BookDetailPage() {
  const { isbn } = useParams()
  if (!isValidString(isbn)) return <div> ISBN is not valid </div>
  return <BookDetailSection isbn={isbn} />
}
```

Reply:
I like that solution, thank you!